python/feathub/dsl/expr_utils.py

```diff
@@ -34,3 +39,34 @@ def get_variables(feathub_expr: str) -> Set[str]:
         if tok.type == "ID" and tok.value not in BUILTIN_FUNC_DEF_MAP:
             variables.add(tok.value)
     return variables
+
+
+def is_id(expr: str) -> bool:
+    """
+    Returns whether the feature expression is a single ID.
+    """
+    match_result = re.match(ID_REGEX, expr)
+    return match_result is not None and match_result.end() == len(expr)
+
+
+def is_static_map_lookup_op(expr: str) -> bool:
+    """
+    Returns whether the feature expression represents a static lookup
+    operation on a map-typed variable. "Static" means the lookup key
+    is a literal instead of a variable.
+    """
+    ast = _parser.parse(expr)
+    return (
+        isinstance(ast, BracketOp)
+        and isinstance(ast.left_child, VariableNode)
+        and isinstance(ast.right_child, ValueNode)
+    )
+
+
+def get_map_static_lookup_key(expr: str) -> Any:
```

Comment:
How about renaming it as `get_static_map_lookup_key` for consistency with `is_static_map_lookup_op`?